saiku-webapp/src/main/resources/ccrs-mondrian-cp1.xml

```diff
@@ -26,8 +26,8 @@
                     <![CDATA[
                         select
                             M.*, CP1.DESIGNATION, CP1.ENTITY_YEAR, CP1.COUNTRY_INCOME_STATUS_ID, CP1.SECONDARY_COUNTRY_INCOME_STATUS_ID,
-                            CASE WHEN cp1.OFFICIAL_REQUEST_DAO_COUNTRY = TRUE THEN 1 ELSE 0 END as OFFICIAL_REQUEST_DAO_COUNTRY,
-                            CASE WHEN M.NON_RESIDENT_AGENCY = TRUE THEN 1 ELSE 0 END as NON_RESIDENT_AGENCY_COUNT
+                            CASE WHEN cp1.OFFICIAL_REQUEST_DAO_COUNTRY = TRUE THEN 1 ELSE CASE WHEN cp1.OFFICIAL_REQUEST_DAO_COUNTRY = FALSE THEN 0 ELSE -1 END END as OFFICIAL_REQUEST_DAO_COUNTRY,
```

Comment:
We're converting here from boolean to integer for counting. This solution will break all measures because it will add 1 and -1 and result will be 0. The idea is that before we had 1 for Yes and 0 for No and Null and count counted only Yes answers. Now nulls will eat from yes answers!

I see that our mondrian.xsd is for older mondrian, pre-3 version. This version does not allow MeasureExpression element. But as far as I saw, our mondrian is v4 and it supports newer v3 schema. It can be accessed here https://github.com/pentaho/mondrian/blob/4.3.0.0-R/src/main/xom/mondrian/olap/Mondrian3Schema.xml.
Please replace our xsd with this newer xsd and hope it works without many changes.
If it works I suggest removing these CASE statements from fact table and use only booleans there. This means that we will have to change dimension table to have key boolean instead of integer. But make sure that null value is visible when you select this dimension in saiku (I think this should work). Also we will have to change measures, because we don't have aggregate function to count true booleans. We can use MeasureExpression tag with `(case COLUMN_NAME when true then 1 else 0 end)`. Please check their example at https://mondrian.pentaho.com/documentation/schema.php#Measures.
The benefit of this approach is that we will use latest documentation which is for v3 and not legacy schema. Easier for us. Also another greater benefit is that all those case statements will just go away hopefully making our queries run faster! Mondrian in this case will only add case expression in specific cases where we measure, i.e. few cases per query. Right now we have to execute all those case statements for each row irrespective if we measure or display that dimension. I think performance will improve.

If changing schema does not work, another solution would be to have two columns for each question. Both use case statements and one uses 1, 0, -1 for dimension while the other uses 1 for true and 0 for rest cases and is used only in measure. I think performance may degrade a bit because number of case statements will double.

Reply:
I just realized that Mondrian3Schema is in XOM format instead of XSD. Also it is strange, because it says that v3 is legacy and the other one is current schema. But their current schema does not have MeasureExpression. If I take a look at mondrian 8 (yes :) v8) they allow MeasureExpression. It appears as if this tag was removed in v4 and the added back later.